### PR TITLE
Issue#5 map for location

### PR DESCRIPTION
**Pull request by harshsharma6401**
### Summary 
For issue #5.  
- Adding Maps and selecting a location.
- Used leaflet.js for adding Map.

###  Author
- [x]  My Pod Leader knows I'm working on this Pull Request
- [x]  I've explained what the Pull Request is adding.
- [x]  I've explained why this is important.

### CODE REVIEW

**Comment by CIPHERTron** [comment]
Screen recording of the feature:

https://user-images.githubusercontent.com/56754747/150298743-163d901c-810d-4012-8fd2-6d77f70ebd64.mov

**Comment by CIPHERTron** [comment]
Heyy @harshsharma6401 this PR is now ready for review. You can now change the state from draft to ready for review 🚀 

Request for review: @gmelodie @ditsuke @Dhruv-Sachdev1313 @Aniket762

**Comment by CIPHERTron** [comment]
Oh okay, just saw that both `package-lock.json` and `yarn.lock` are present. I guess Harsh uses `npm`
Also, noticed that there are two extra dependencies i.e. `react-geolocated` and `typescript` which I don't think we are using them anywhere
Will make the necessary changes and push them 💪🏻

**Comment by harshsharma6401** [comment]
> Oh okay, just saw that both `package-lock.json` and `yarn.lock` are present. I guess Harsh uses `npm` Also, noticed that there are two extra dependencies i.e. `react-geolocated` and `typescript` which I don't think we are using them anywhere Will make the necessary changes and push them 💪🏻

I installed Eslint extension, so typescript was its unmet dependency. That is why I installed it.

File: package.json [hunk 1]
```diff
@@ -23,16 +26,12 @@
...
-      ">0.2%",
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
+  "browserslist": [
```
Comment (ditsuke) [inline package.json:29]: Please revert the changes to the "browserslist" key

**Review by ditsuke — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (harshsharma6401) [reply package.json:29]: This was causing problems for Map to display. It was the only solution to get rid of the errors. 
The development part "last 1 chrome version", .... was actually the problem so we had to eliminate it.

File: src/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
[the remark this answers is not in the document]
Comment (harshsharma6401) [reply src/.editorconfig]: Yeah, this is the same one !

File: package.json (hunk 1 above)
Comment (CIPHERTron) [reply package.json]: Done

Comment (CIPHERTron) [reply package.json]: Okay, that seems the problem
I guess I have to revert it again cuz now it's breaking :/

Comment (harshsharma6401) [reply package.json]: > Okay, that seems the problem I guess I have to revert it again cuz now it's breaking :/

LGTM

Comment (CIPHERTron) [reply package.json:29]: > LGTM

It might be your `node_modules` cache 🤔

**Comment by CIPHERTron** [comment]
Also, do we require approval from MLH to deploy the preview of this PR?

cc: @wrussell1999 @gmelodie

File: package.json (hunk 1 above)
Comment (harshsharma6401) [reply package.json:29]: You have to delete  .cache file of node_modules.  Then it'll work again.

**Comment by yyjlincoln** [comment]
> Also, do we require approval from MLH to deploy the preview of this PR?
> 
> cc: @wrussell1999 @gmelodie

Yeah, I think so. Only collaborators can deploy the preview I think.

File: src/App.js [hunk 3]
```diff
@@ -1,54 +1,83 @@
...
-          onChange={event => setCity(event.target.value)} />
-        <div className="Results">
-          {!isLoaded && <h2>Loading...</h2>}
-          {console.log(results)}
-          {isLoaded && results && <>
-            <h3>{results.weather[0].main}</h3>
-            <p>Feels like {results.main.feels_like}°C</p>
-            <i><p>{results.name}, {results.sys.country}</p></i>
-          </>}
+    return (
```
Comment (ditsuke) [inline src/App.js:45]: Can you revert changes other than the additions here? I volunteer to help if wanted!

File: src/App.js [hunk 4]
```diff
@@ -1,54 +1,83 @@
...
-import logo from './mlh-prep.png'
+import "./App.css";
+import logo from "./mlh-prep.png";
+import WeatherMap from "./components/WeatherMap";
 
 function App() {
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
-  const [city, setCity] = useState("New York City")
+  const [city, setCity] = useState("New York City");
```
Comment (ditsuke) [inline src/App.js:9]: Revert the change on this line please. We'll do a single PR for all styling fixes once #17 is merged in.

File: src/App.js [hunk 5]
```diff
@@ -1,54 +1,83 @@
...
   const [results, setResults] = useState(null);
+  const [cityCoordinates, setCityCoordinates] = useState({
+    lat: "51.505",
+    lon: "-0.09",
+  });
 
   useEffect(() => {
-    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${process.env.REACT_APP_APIKEY}`)
-      .then(res => res.json())
+    fetch(
```
Comment (ditsuke) [inline src/App.js:17]: revert change

File: src/App.js [hunk 6]
```diff
@@ -1,54 +1,83 @@
...
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${process.env.REACT_APP_APIKEY}`
+    )
+      .then((res) => res.json())
       .then(
         (result) => {
-          if (result['cod'] !== 200) {
-            setIsLoaded(false)
+          if (result["cod"] !== 200) {
```
Comment (ditsuke) [inline src/App.js:23]:
```suggestion
          if (result['cod'] !== 200) {
```

File: src/App.js [hunk 7]
```diff
@@ -1,54 +1,83 @@
...
+            setIsLoaded(false);
```
Comment (ditsuke) [inline src/App.js:24]:
```suggestion
            setIsLoaded(false)
```

File: src/App.js [hunk 8]
```diff
@@ -1,54 +1,83 @@
...
           }
         },
         (error) => {
           setIsLoaded(true);
           setError(error);
         }
-      )
-  }, [city])
+      );
+  }, [city]);
```
Comment (ditsuke) [inline src/App.js:40]:
```suggestion
      )
  }, [city])
```

this is correct but lets revert for now

**Review by ditsuke — CHANGES_REQUESTED** [review]

**Comment by CIPHERTron** [comment]
@ditsuke that's because of my global prettier configuration I guess. Let me disable prettier for this project as of now and revert the changes :)

**Comment by CIPHERTron** [comment]
All the conflicts are resolved and this PR is good to go 🚀

**Comment by ditsuke** [comment]
I found a small issue with 0d22719. I'll push a fix along with some refactoring.

recording

**Comment by ditsuke** [comment]
Okay, this is looking great to me after 57be5f4. I'd like to squash some commit ranges before approving!

**Comment by ditsuke** [comment]
This was a major PAIN to rebase but I ended up learning so much about interactive rebasing. TIP: the relatively new `--rebase-merges` flag.

**Review by Dhruv-Sachdev1313 — APPROVED** [review]
This Looks Great!

**Comment by ditsuke** [comment]
RTC and no conflicts. Merging this in!

**Comment by ditsuke** [comment]
Great work guys

**Pull request merged**